Brokerages/BrokerageAuthentication.cs

```diff
@@ -0,0 +1,34 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Orders;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Brokerages
+{
+    public abstract class BrokerageAuthentication
+    {
+        /// <summary>
+        /// Validate Brokerage Authentication Parameters
+        /// </summary>
+        /// <returns>true for OK (ie. no error)</returns>
+        public abstract bool Validate();
```

Comment:
We may need a way of communicating what failed validation. We could possible have this return a [ValidationResult](https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.validationresult%28v=vs.110%29.aspx) instance instead of just true/false.